hugr/src/types/type_param.rs

```diff
@@ -136,6 +136,30 @@ impl From<Type> for TypeArg {
     }
 }
 
+impl From<CustomTypeArg> for TypeArg {
+    fn from(arg: CustomTypeArg) -> Self {
+        Self::Opaque { arg }
+    }
+}
+
+impl From<TypeArgVariable> for TypeArg {
```

Comment:
Originally IIRC the plan was for this struct to be somewhat opaque, and one would use `TypeArg::new_var_use` to construct one. So do we really want/need this?

Other than that, all looks good to me, thanks! :-)